docs/driver-only-builds.md

```diff
@@ -238,37 +240,121 @@ The same holds for the associated algorithm:
-
-[This section might contain incomplete data and it is going to be updated in
-#8358, i.e. the wrap-up task for accelerated ciphers and AEADs.]
-
-It is possible to have all AEADs operations provided only by a driver.
-
-More precisely you can:
-- enable desired PSA algorithm(s) and key type(s):
-  - `PSA_WANT_ALG_[CCM|GCM]` with `PSA_WANT_KEY_TYPE_[AES|ARIA|CAMELLIA]`
-  - `PSA_WANT_ALG_CHACHA20_POLY1305` with `PSA_WANT_KEY_TYPE_CHACHA20`;
-- enable `MBEDTLS_PSA_ACCEL_xxx` symbol(s) which correspond to the
-  `PSA_WANT_xxx` of the previous step (both for algorithms and key types);
-- disable builtin support of `MBEDTLS_[CCM|GCM|CHACHAPOLY|POLY1305]_C`
-  algorithms and key types `MBEDTLS_[AES|ARIA|CAMELLIA|CHACHA20]_C` for AEADs
-  which are accelerated.
-
-In a build in which all AEADs algorithms and related key types are accelerated
-all AEADs operations requested through the PSA Crypto API (including those in
-TLS and X.509) will be performed by the driver.
-Moreover if no unauthenticated cipher is required, it is also possible to
-disable all built-in block cipher's key types
-(i.e. `MBEDTLS_[AES|ARIA|CAMELLIA|CHACHA20]_C`) and `MBEDTLS_CIPHER_C`. This
-helps in further reducing code's footprint, but unfortunately it makes the
-following modules unavailable:
-- `MBEDTLS_PKCS[5|12]_C`
-- `MBEDTLS_CTR_DRBG_C`
-- `MBEDTLS_NIST_KW_C`
+Ciphers and AEADs
+-----------------
+
+It is possible to have all ciphers and AEADs operations provided only by a
+driver. More precisely, for each desired combination of key type and
+algorithm/mode you can:
+- enable desired PSA key type(s):
+  - `PSA_WANT_KEY_TYPE_AES`,
+  - `PSA_WANT_KEY_TYPE_ARIA`,
+  - `PSA_WANT_KEY_TYPE_CAMELLIA`,
+  - `PSA_WANT_KEY_TYPE_CHACHA20`,
+  - `PSA_WANT_KEY_TYPE_DES`;
+- enable desired PSA algorithm(s):
+  - unauthenticated ciphers modes:
+    - `PSA_WANT_ALG_CBC_NO_PADDING`,
+    - `PSA_WANT_ALG_CBC_PKCS7`,
+    - `PSA_WANT_ALG_CCM_STAR_NO_TAG`,
+    - `PSA_WANT_ALG_CFB`,
+    - `PSA_WANT_ALG_CTR`,
+    - `PSA_WANT_ALG_ECB_NO_PADDING`,
+    - `PSA_WANT_ALG_OFB`,
+    - `PSA_WANT_ALG_STREAM_CIPHER`;
+  - AEADs:
+    - `PSA_WANT_ALG_CCM`,
+    - `PSA_WANT_ALG_GCM`,
+    - `PSA_WANT_ALG_CHACHA20_POLY1305`;
+- enable `MBEDTLS_PSA_ACCEL_[KEY_TYPE_xxx|ALG_yyy]` symbol(s) which correspond
+   to the PSA_WANT_KEY_TYPE_xxx` and `PSA_WANT_ALG_yyy` of the previous steps;
+- disable builtin support of key types:
+  - `MBEDTLS_AES_C`,
+  - `MBEDTLS_ARIA_C`,
+  - `MBEDTLS_CAMELLIA_C`,
+  - `MBEDTLS_DES_C`,
+  - `MBEDTLS_CHACHA20_C`;
+  and algorithms/modes:
+  - `MBEDTLS_CBC_C`
+  - `MBEDTLS_CFB_C`
+  - `MBEDTLS_CTR_C`
+  - `MBEDTLS_OFB_C`
+  - `MBEDTLS_XTS_C`
+  - `MBEDTLS_CCM_C`
+  - `MBEDTLS_GCM_C`
+  - `MBEDTLS_CHACHAPOLY_C`
+  - `MBEDTLS_NULL_CIPHER`
+
+Once a key type and related algorithm are accelerated, all cipher and AEADs
+operations of that type requested through the PSA Crypto API are performed by
+the driver. Only functions belonging to legacy modules which are disabled won't
```

Comment:
The last sentence is not correct. Functions like `mbedtls_pkcs5_pbes2_ext()` are not going to be available either. Also, some functions are available but with a reduced feature set.

I think we should replace this sentence with "All the PSA Crypto APIs will work, as well as X.509 and TLS (with `MBEDTLS_USE_PSA_CRYPTO` enabled) but some non-PSA APIs will be absent or have reduced functionality, see the next section for details".

In order to remember the full list of details, we can have a look at ignored things for `analyze_driver_vs_reference_cipher_aead` in `analyze_outcomes.py` plus things that are disabled in the reference component. So, if I'm not missing anything:
- PKCS5/PKCS12 en/decryption won't work (key derivations function from those modules still work).
- Parsing encrypted PEM won't work.
- Parsing of encrypted keys (PEM or PKCS12) in PK parse won't work.
- CMAC will not be able to use AES if `AES_C` is disabled (could still work with DES)*.
- NIST-KW needs to be disabled if `AES_C` is disabled.

Actually looking again at those test components, I'm no longer sure disabling `CIPHER_C` in the accelerated component was the right move. Perhaps we should have two accelerated components: one with `CIPHER_C` disabled and one with it enabled - to ensure that the parts of PKCS5/12 that are suppose to work indeed do.

Or perhaps instead we should just update the dependencies of PKCS5/12? We don't need to have the whole modules depending on `CIPHER_C`, but only the functions that are about encryption.

*We should not ignore the whole CMAC test suite in `analyze_driver_vs_reference_cipher_aead`, only the cases that use AES.

Reply:
> Or perhaps instead we should just update the dependencies of PKCS5/12? We don't need to have the whole modules depending on CIPHER_C, but only the functions that are about encryption.

Sounds good! I will add this.

> We should not ignore the whole CMAC test suite in analyze_driver_vs_reference_cipher_aead, only the cases that use AES.

Not sure about this. `analyze_driver_vs_reference_cipher_aead` uses `test_psa_crypto_config_accel_cipher_aead` as accelerated component and this, as its name suggests, accelerates *all* ciphers and AEADs. Therefore not only `AES_C` is disabled in the accelerated component, but also `DES_C`. As a consequence *all* test of `test_suite_cmac` are actually to be ignored. Am I missing something?

Reply:
Nope, you not missing anything, I was - for some reason I forgot `DES_C` was disabled as well. So, please ignore that part of my comment and sorry for the noise.

Reply:
Though I'll note the component's name is slightly misleading as it says `cipher_aead` but it also accelerates `cmac` which is none of these (as the name implies, it's a Message Authentication Code). Should we rename to `accel_cipher_aead_cmac`? (Makes sense as `accel` components are grouped by families of key types, and ciphers, AEADs and CMAC all use the same family of key types.)

Reply:
I think it would really be clearer to rename to `accel_cipher_aead_cmac`, can you do it?